docs/evaluation/speech-audio.md

```diff
@@ -0,0 +1,215 @@
+# Speech and Audio Understanding
+
+This section details how to evaluate speech and audio understanding benchmarks, testing models' ability to understand and reason about audio content including speech, music, and environmental sounds.
+
+## Supported benchmarks
+
+### MMAU-Pro
+
+MMAU-Pro (Multimodal Audio Understanding - Pro) is a comprehensive benchmark for evaluating audio understanding capabilities across three different task categories:
+
+- **Closed-form questions**: Questions with specific answers evaluated using NVEmbed similarity matching
+- **Open-ended questions**: Questions requiring detailed responses, evaluated with LLM-as-a-judge (Qwen 2.5)
+- **Instruction following**: Tasks that test the model's ability to follow audio-related instructions
+
+#### Dataset Location
+
+- Benchmark is defined in [`nemo_skills/dataset/mmau-pro/__init__.py`](https://github.com/NVIDIA-NeMo/Skills/blob/main/nemo_skills/dataset/mmau-pro/__init__.py)
+- Original benchmark source is hosted on [HuggingFace](https://huggingface.co/datasets/gamma-lab-umd/MMAU-Pro)
+
+## Preparing MMAU-Pro Data
+
+MMAU-Pro requires audio files for evaluation. The `--with-audio` parameter controls whether audio files are downloaded.
+
+> **Note:** Meaningful evaluation requires audio files and audio-capable models. Text-only data preparation is possible but not recommended.
+
+### Data Preparation with Audio Files
+
+To prepare the dataset with audio files:
+
+```bash
+export HF_TOKEN=your_huggingface_token
+ns prepare_data mmau-pro --with-audio
+```
+
+**What happens:**
+1. Downloads audio archive (~50GB) from HuggingFace
+2. Requires authentication (HuggingFace token via `HF_TOKEN` environment variable)
+3. Extracts audio files to dataset directory
+
+
+### Custom Data Directory
+
+To store the dataset in a specific location:
+
+```bash
+ns prepare_data mmau-pro --with-audio --data-dir=/path/to/mmau-pro-data
+```
+
+Useful for cluster storage, sharing data across jobs, or persistent storage across runs.
+
+## Running Evaluation
+
+> **⚠️ Warning:** MMAU-Pro evaluation currently supports only the Megatron server type (`--server_type=megatron`) with audio-capable models. Support for additional server types is planned for future releases.
+
+### Evaluation Example
+
+Complete example using Megatron-based audio-visual language models:
+
+```bash
+# Set up environment variables - ADJUST THESE PATHS TO YOUR SETUP
+export MEGATRON_PATH="/workspace/path/to/megatron-lm"
+export CKPT_PATH=/workspace/path/to/checkpoint-tp1
+export MODEL_CFG_PATH="${CKPT_PATH}/config.yaml"
+export SERVER_ENTRYPOINT="$MEGATRON_PATH/path/to/server.py"
+export SERVER_CONTAINER="/path/to/server_container.sqsh"
+export OUTPUT_DIR="mmau-pro-eval"
+
+# Set up keys
+export HF_TOKEN='your_huggingface_token'
+export NVIDIA_API_KEY='your_nvidia_api_key'
+export WANDB='your_wandb_key'   # not neccessary
+
+# Run evaluation with audio support
+export HF_TOKEN=${HF_TOKEN} && \
```

Comment:
these 3 lines are redundant and shouldn't be necessary as you already export above